ostd/src/arch/x86/tdx_guest.rs

```diff
@@ -100,6 +94,14 @@ pub unsafe fn protect_gpa_range(gpa: Paddr, page_num: usize) -> Result<(), PageC
         warn!("Misaligned address: {:x}", gpa);
     }
 
+    if gpa + page_num * PAGE_SIZE > crate::mm::frame::max_paddr() {
+        warn!(
+            "GPA range [{:#x}, {:#x}) is not within the linear mapping",
```

Comment:
I suppose that you want to print the warning in case that this function is misused by `IoMem` in the future when it implements the `Drop` trait. But why not make the function right so that it can't be misused? The protect function is symmetric to the unprotect function. So I suppose the protect function should be refactored in the same way as the unprotect function.